pyvista/core/pointset.py

```diff
@@ -142,50 +197,74 @@ class PolyData(_vtk.vtkPolyData, PointSet, PolyDataFilters):
+                verts = self._make_vertex_cells(self.n_points)
+                self.verts = CellArray(verts, self.n_points, deep)
 
-        if not args:
             return
-        elif len(args) == 1:
-            if isinstance(args[0], _vtk.vtkPolyData):
-                if deep:
-                    self.deep_copy(args[0])
-                else:
-                    self.shallow_copy(args[0])
-            elif isinstance(args[0], (str, pathlib.Path)):
-                self._from_file(args[0])
-            elif isinstance(args[0], (np.ndarray, list)):
-                if isinstance(args[0], list):
-                    points = np.asarray(args[0])
-                    if not np.issubdtype(points.dtype, np.number):
-                        raise TypeError('Points must be a numeric type')
-                else:
-                    points = args[0]
-                if points.ndim != 2:
-                    points = points.reshape((-1, 3))
-                cells = self._make_vertex_cells(points.shape[0])
-                self._from_arrays(points, cells, deep, verts=True)
-            else:
-                raise TypeError('Invalid input type')
 
-        elif len(args) == 2:
-            arg0_is_array = isinstance(args[0], np.ndarray)
-            arg1_is_array = isinstance(args[1], np.ndarray)
-            if arg0_is_array and arg1_is_array:
-                self._from_arrays(args[0], args[1], deep)
+        # PolyData-like
+        elif isinstance(var_inp, (PolyData, _vtk.vtkPolyData)):
+            for kwarg in ['faces', 'n_faces', 'lines']:
+                if local_parms[kwarg]:
+                    raise ValueError('No other arguments should be set when first '
+                                     'parameter is a PolyData')
+            if deep:
+                self.deep_copy(var_inp)
             else:
-                raise TypeError('Invalid input type')
-        else:
-            raise TypeError('Invalid input type')
+                self.shallow_copy(var_inp)
+            return
 
-        # Check if need to make vertex cells
-        if self.n_points > 0 and self.n_cells == 0:
-            # make vertex cells
-            self.verts = self._make_vertex_cells(self.n_points)
+        # First parameter is points
+        elif isinstance(var_inp, (np.ndarray, list)):
+            points = var_inp
+            if not isinstance(points, np.ndarray):
+                points = np.array(points)
+            self.SetPoints(pyvista.vtk_points(points, deep=deep))
+        else:
+            TypeError('Invalid Input type:\n\n'
+                      'Expected first argument to be either a:\n'
+                      '    - vtk.PolyData\n'
+                      '    - pyvista.PolyData\n'
+                      '    - numeric numpy.ndarray (1 or 2 dimensions)\n'
+                      '    - List (flat or nested with 3 points per vertex)\n\n'
+                      f'Instead got: {type(var_inp)}')
```

Comment:
Completely subjective alternate suggestion, swapping the quote marks and linefeeds to different kind of fluff:
```py
from textwrap import dedent
[...]
            msg = f"""
                Invalid Input type:
             
                Expected first argument to be either a:
                - vtk.PolyData
                - pyvista.PolyData
                - numeric numpy.ndarray (1 or 2 dimensions)
                - List (flat or nested with 3 points per vertex)
                               
                Instead got: {type(var_inp)}"""
            raise TypeError(dedent(msg.strip('\n')))
```

Reply:
Implemented it.  This is way cleaner and completely removes the need to have `'` when trying to keep the same indentation.  This is a really good practice, makes it much easier to write long errors provided you don't need indents.

Reply:
It would also preserve indents if present between lines. `textwrap.dedent` only removes the longest indent common to all lines.